pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -732,20 +727,25 @@ func (m *kubeGenericRuntimeManager) doPodResizeAction(pod *v1.Pod, podStatus *ku
 			result.Fail(err)
 			return
 		}
-		if currentPodMemoryUsage >= uint64(*podResources.Memory) {
+		if podResources.Memory != nil && currentPodMemoryUsage >= uint64(*podResources.Memory) {
 			klog.ErrorS(nil, "Aborting attempt to set pod memory limit less than current memory usage", "pod", pod.Name)
 			result.Fail(fmt.Errorf("Aborting attempt to set pod memory limit less than current memory usage for pod %s", pod.Name))
 			return
 		}
-		if errResize := resizeContainers(v1.ResourceMemory, int64(*currentPodMemoryConfig.Memory), *podResources.Memory, 0, 0); errResize != nil {
+		var currentMemory, desiredMemory int64
+		if podResources.Memory != nil {
+			currentMemory = *currentPodMemoryConfig.Memory
+			desiredMemory = *podResources.Memory
+		}
+		if errResize := resizeContainers(v1.ResourceMemory, currentMemory, desiredMemory, 0, 0); errResize != nil {
```

Comment:
In this PR, if `podResources.Memory` is `nil`, `writeCgroupMemoryLimit()` is not called because both `currentMemory` and `desiredMemory` are zero. I will consider this at #127143.